### PR TITLE
wrote tests for team views functions

### PR DESCRIPTION
**Pull request by VasilHamulyak**
## JIRA

* [Main JIRA ticket](https://ssu-jira.softserveinc.com/browse/LVPYTP-133)

## Code reviewers

- [ ] @dmytrorevak
- [ ] @IvanAndrukh
- [ ] @JjuliaP
- [ ] @makarman
- [ ] @noobhunters
- [ ] @Silent777
- [ ] @Staryj812

### Second Level Review

- [ ] @ssOleg
- [ ] @lhalam

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: eventually/tests/unittests/utils/testteamviewsfunctions.py [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+                'description': 'description',
+                'image': 'link',
+                'owner_id': 20,
+                'members_id_add': ['not int'],
+                'members_id_del': []
+                }
+
+        team_update_dict = team_views_functions.update_team_dict(data, user)
+
+        self.assertIsNone(team_update_dict)
```
Comment (Silent777) [inline eventually/tests/unittests/utils/testteamviewsfunctions.py]: Add end line

**Review by Silent777 — DISMISSED** [review]

File: eventually/tests/unittests/utils/testteamviewsfunctions.py (hunk 1 above)
Comment (redmyt) [reply eventually/tests/unittests/utils/testteamviewsfunctions.py]: Remove binary files from commit

File: eventually/tests/unittests/utils/testteamviewsfunctions.py [hunk 2]
```diff
@@ -0,0 +1,270 @@
+from django.test import TestCase
+from authentication.models import CustomUser
+from utils import team_views_functions
+
+
+class UtilsTeamViewsFunctionsTestCase(TestCase):
+
+    def setUp(self):
+
+            CustomUser(
```
Comment (redmyt) [inline eventually/tests/unittests/utils/testteamviewsfunctions.py]: use python format style

File: eventually/tests/unittests/utils/testteamviewsfunctions.py [hunk 3]
```diff
@@ -0,0 +1,270 @@
+from django.test import TestCase
+from authentication.models import CustomUser
+from utils import team_views_functions
+
+
+class UtilsTeamViewsFunctionsTestCase(TestCase):
```
Comment (redmyt) [inline eventually/tests/unittests/utils/testteamviewsfunctions.py]: maybe you can use the more meaningful name?

File: eventually/tests/unittests/utils/testteamviewsfunctions.py [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+                       email='email@email.com',
+                       password='1111').save()
+
+            CustomUser(
+                       id=200,
+                       email='tets@email.com',
+                       password='0000').save()
+
+    def test_find_users_by_id_success(self):
+        members_list = team_views_functions.find_users([20])
```
Comment (redmyt) [inline eventually/tests/unittests/utils/testteamviewsfunctions.py]: it is good practice to name this kind of data something like `actual_members_list` or `current_members_list`

File: eventually/tests/unittests/utils/testteamviewsfunctions.py [hunk 5]
```diff
@@ -0,0 +1,288 @@
...
+
+        members_list = team_views_functions.find_users([25])
+
+        self.assertIsNone(members_list)
+
+    def test_create_team_dict_success(self):
+        """Method that tests succeeded `create_team_dict` method."""
+
+        user = CustomUser.objects.get(id=20)
+        data = {
```
Comment (redmyt) [inline eventually/tests/unittests/utils/testteamviewsfunctions.py:43]: use python code style

File: eventually/tests/unittests/utils/testteamviewsfunctions.py [hunk 6]
```diff
@@ -0,0 +1,288 @@
...
+               'name': 'name',
+               'description': 'description',
+               'image': 'link',
+               'owner': 20,
+               'members': [20]
+               }
```
Comment (redmyt) [inline eventually/tests/unittests/utils/testteamviewsfunctions.py]: use python code style

File: eventually/tests/unittests/utils/testteamviewsfunctions.py [hunk 7]
```diff
@@ -0,0 +1,288 @@
...
+
+        expect_team_dict = {
```
Comment (redmyt) [inline eventually/tests/unittests/utils/testteamviewsfunctions.py]: use python code style

File: eventually/tests/unittests/utils/testteamviewsfunctions.py [hunk 8]
```diff
@@ -0,0 +1,288 @@
...
+
+        team_dict = team_views_functions.create_team_dict(data, user)
+
+        self.assertDictEqual(team_dict, expect_team_dict)
+
+    def test_create_team_dict_not_success_value_list_of_int_validator(self):
+        """Method that tests unsucceeded `create_team_dict` method if value members_id not valid"""
+
+        user = CustomUser.objects.get(id=20)
+        data = {
```
Comment (redmyt) [inline eventually/tests/unittests/utils/testteamviewsfunctions.py:67]: use python code style

File: eventually/tests/unittests/utils/testteamviewsfunctions.py [hunk 9]
```diff
@@ -0,0 +1,288 @@
...
+
+        team_dict = team_views_functions.create_team_dict(data, user)
+
+        self.assertIsNone(team_dict)
+
+    def test_create_team_dict_not_success_if_not_members(self):
+        """Method that tests unsucceeded `create_team_dict` method if members_id not exist"""
+
+        user = CustomUser.objects.get(id=20)
+        data = {
```
Comment (redmyt) [inline eventually/tests/unittests/utils/testteamviewsfunctions.py:83]: use python code style

File: eventually/tests/unittests/utils/testteamviewsfunctions.py [hunk 10]
```diff
@@ -0,0 +1,288 @@
...
+
+        get_users_tuple = team_views_functions.get_users(data, user)
+
+        self.assertTupleEqual(get_users_tuple, expect_tuple)
+
+    def test_get_users_not_add_members_id_add_is_empty(self):
+        """Method that tests unsucceeded `get_users` method members_id_add empty"""
+
+        user = CustomUser.objects.get(id=20)
+        data = {
```
Comment (redmyt) [inline eventually/tests/unittests/utils/testteamviewsfunctions.py:117]: use python code style

File: eventually/tests/unittests/utils/testteamviewsfunctions.py [hunk 11]
```diff
@@ -0,0 +1,288 @@
...
+
+        get_users_tuple = team_views_functions.get_users(data, user)
+
+        self.assertTupleEqual(get_users_tuple, expect_tuple)
+
+    def test_get_users_not_add_member_owner_get_bad_value(self):
+        """Method that tests unsucceeded `create_team_dict` method if members_id not exist"""
+
+        user = CustomUser.objects.get(id=20)
+        data = {
```
Comment (redmyt) [inline eventually/tests/unittests/utils/testteamviewsfunctions.py]: use python code style

File: eventually/tests/unittests/utils/testteamviewsfunctions.py [hunk 12]
```diff
@@ -0,0 +1,288 @@
...
+
+        expect_tuple = (None, None, None)
+
+        get_users_tuple = team_views_functions.get_users(data, user)
+
+        self.assertTupleEqual(get_users_tuple, expect_tuple)
+
+    def test_get_users_not_add_member_members_id_add_bad_value(self):
+        user = CustomUser.objects.get(id=20)
+        data = {
```
Comment (redmyt) [inline eventually/tests/unittests/utils/testteamviewsfunctions.py]: use python code style

File: eventually/tests/unittests/utils/testteamviewsfunctions.py [hunk 13]
```diff
@@ -0,0 +1,288 @@
...
+
+        get_users_tuple = team_views_functions.get_users(data, user)
+
+        self.assertIsNone(get_users_tuple)
+
+
+    def test_get_users_delete_members_success(self):
+        user = CustomUser.objects.get(id=20)
+        members_del = CustomUser.objects.get(id=200)
+        data = {
```
Comment (redmyt) [inline eventually/tests/unittests/utils/testteamviewsfunctions.py]: use python code style

File: eventually/tests/unittests/utils/testteamviewsfunctions.py [hunk 14]
```diff
@@ -0,0 +1,288 @@
...
+
+        expect_tuple = (user, [members_del], None)
+
+        get_users_tuple = team_views_functions.get_users(data, user)
+
+        self.assertTupleEqual(get_users_tuple, expect_tuple)
+
+    def test_get_users_delete_members_not_success_member_not_in_list(self):
+        user = CustomUser.objects.get(id=20)
+        data = {
```
Comment (redmyt) [inline eventually/tests/unittests/utils/testteamviewsfunctions.py]: use python code style

File: eventually/tests/unittests/utils/testteamviewsfunctions.py [hunk 15]
```diff
@@ -0,0 +1,288 @@
...
+                'members_id_del': [201]
+                }
+
+        get_users_tuple = team_views_functions.get_users(data, user)
+
+        self.assertIsNone(get_users_tuple)
+
+    def test_get_users_delete_members_not_success_member_is_user(self):
+        user = CustomUser.objects.get(id=20)
+        data = {
```
Comment (redmyt) [inline eventually/tests/unittests/utils/testteamviewsfunctions.py]: use python code style

File: eventually/tests/unittests/utils/testteamviewsfunctions.py [hunk 16]
```diff
@@ -0,0 +1,288 @@
...
+                'members_id_del': [20]
+                }
+
+        get_user_tuple = team_views_functions.get_users(data, user)
+
+        self.assertIsNone(get_user_tuple)
+
+    def test_get_users_delete_members_not_success_members_id_del_bad_value(self):
+        user = CustomUser.objects.get(id=20)
+        data = {
```
Comment (redmyt) [inline eventually/tests/unittests/utils/testteamviewsfunctions.py]: use python code style

File: eventually/tests/unittests/utils/testteamviewsfunctions.py [hunk 17]
```diff
@@ -0,0 +1,288 @@
...
+                }
+
+        get_user_tuple = team_views_functions.get_users(data, user)
+
+        self.assertIsNone(get_user_tuple)
+
+    def test_update_team_dict_success(self):
+        user = CustomUser.objects.get(id=20)
+        member = CustomUser.objects.get(id=200)
+        data = {
```
Comment (redmyt) [inline eventually/tests/unittests/utils/testteamviewsfunctions.py]: use python code style

File: eventually/tests/unittests/utils/testteamviewsfunctions.py [hunk 18]
```diff
@@ -0,0 +1,288 @@
...
+
+
+        team_update_dict = team_views_functions.update_team_dict(data, user)
+
+        self.assertDictEqual(team_update_dict, expect_dict)
+
+
+    def test_update_team_dict_not_success_image_not_str(self):
+        user = CustomUser.objects.get(id=20)
+        data = {
```
Comment (redmyt) [inline eventually/tests/unittests/utils/testteamviewsfunctions.py]: use python code style

**Review by lhalam — APPROVED** [review]

**Pull request merged**